docs/modules/ROOT/pages/dapp.adoc

```diff
@@ -0,0 +1,337 @@
+✓ Instance created at 0xCfEB869F69431e42cdB54A4F4f105C19C080A601
+0xCfEB869F69431e42cdB54A4F4f105C19C080A601
+----
+
+== Creating the dapp
+We will create our dapp using the `create-react-app` package, which bootstraps a simple client-side application using React.
+
+[source,console]
+----
+$ npx create-react-app client
+----
+
+== Setup Solidity Hot Loader
+Now that the directories for our dapp have been created, we can install Solidity Hot Loader. In the `client` directory install *Solidity Hot Loader* and `json-loader`.
+
+[source,console]
+----
+$ npm install @openzeppelin/solidity-loader json-loader
+----
+
+Next up, we configure Solidity Hot Loader and set it to enabled. To stop the Solidity Hot Loader automagically updating our contracts, set disabled to `true`.
+
+In the `client` directory create a `config` directory. In the `config` directory create a `webpack.js` file for the Solidity Hot Loader webpack rule with the following code, notice that we are starting with the Solidity Hot Loader enabled (`disabled: false`):
+
+[source,javascript]
+----
+const solidityLoaderOptions = {
+  network: 'development',
+  // you can stop loader from automatic compile/push/upgrade
+  // action by setting disabled flag to true, but it will still
+  // serve .json files from file system
+  disabled: false,
+};
+
+module.exports = {
+  solidityLoader: {
+    test: /\.sol$/,
+    use: [
+      { loader: 'json-loader' },
+      {
+        loader: '@openzeppelin/solidity-loader',
+        options: solidityLoaderOptions,
+      },
+    ],
+  },
+  solidityLoaderOptions,
+};
+----
+
+We now add in the Solidity Hot Loader to webpack.  In the `client` directory create a `config-overrides.js` file with the following code:
+
+[source,javascript]
+----
+/* config-overrides.js */
+const { solidityLoader } = require('./config/webpack');
+
+module.exports = function override(config, env) {
+  //do stuff with the webpack config...
+
+  // allow importing from outside of app/src folder, ModuleScopePlugin prevents this.
+  const scope = config.resolve.plugins.findIndex(o => o.constructor.name === 'ModuleScopePlugin');
+  if (scope > -1) {
+    config.resolve.plugins.splice(scope, 1);
+  }
+
+  // add solidity loader support
+  // have to insert before last loader, because CRA user 'file-loader' as default one
+  config.module.rules.splice(config.module.rules - 2, 0, solidityLoader);
+
+  return config;
+};
+----
+
+In this guide we use https://github.com/timarney/react-app-rewired[`react-app-rewired`] to rewire `create-react-app` webpack configs to add in the Solidity Hot Loader without ejecting from `create-react-app`.
+
+In the `client` directory, install `react-app-rewired`
+
+[source,console]
+----
+$ npm i react-app-rewired
+----
+
+To use `react-app-rewired`, we now need to change the `package.json` scripts.  In `package.json` in the `client` directory, update the scripts to use `react-app-rewired` so that they looks as follows:
+
+[source,javascript]
```

Comment:
I think this may read better as a `diff`

Reply:
Changed to `diff`